core/controllers/reader.py

```diff
@@ -418,14 +436,57 @@ class ExplorationMaybeLeaveHandler(base.BaseHandler):
     @require_playable
     def post(self, exploration_id):
         """Handles POST requests."""
+        version = self.payload.get('version')
+        state_name = self.payload.get('state_name')
+        user_id = self.user_id
+        timestamp = datetime.datetime.utcnow()
+        collection_id = self.payload.get('collection_id')
+
+        if user_id:
+            learner_progress_services.add_exp_to_incomplete_list(
```

Comment:
I have addressed this in `add_exp_to_incomplete_list`. If the exploration has already been completed it would not be added to the incomplete list.